app/modules/topics/components/Editor.js

```diff
@@ -26,36 +39,74 @@ type StateProps = {
   topic: Topic,
 };
 
-type Props = CustomTranslatorProps & PassedProps & StateProps;
+type Props = CustomTranslatorProps & DispatchProps & PassedProps & StateProps;
 
 const mapStateToProps = (state: State, props: PassedProps): StateProps => {
   const topic = getById(state, { id: props.topicId });
 
-  if (topic == null) {
-    throw new ObjectNotFoundError('topics:topic', props.topicId);
-  }
-
   return {
     topic,
   };
 };
 
+const mapDispatchToProps = (dispatch: Dispatch<*>): DispatchProps => {
+  return {
+    onSaveButtonClick: (id: Identifier): void => {
+      dispatch(save(id));
+    },
+    onLoadButtonClick: (id: Identifier): void => {
+      dispatch(load(id));
+    },
+  };
+};
+
 const ContentItemEditableDisplay = contentItems.components.EditableDisplay;
 
 const PureEditor = (props: Props): React.Node => {
-  const { topic } = props;
+  const {
+    t,
+    topicId,
+    topic,
+    onSaveButtonClick,
+    onLoadButtonClick,
+  } = props;
+
+  if (!topic) {
+    onLoadButtonClick(topicId);
+
+    return (
+      <div>
+        <ApiDimmer request={API_GET_TOPIC_CONTENT}>{t('editor:api.load.pending')}</ApiDimmer>
+      </div>
+    );
+  }
 
   return (
     <div>
       <Header as="h1">{topic.title}</Header>
+
+      <FlashMessages />
+
+      <ApiDimmer request={API_GET_TOPIC_CONTENT}>{t('editor:api.load.pending')}</ApiDimmer>
+      <ApiDimmer request={API_PATCH_TOPIC_CONTENT}>{t('editor:api.save.pending')}</ApiDimmer>
+
+      <p>
+        <Button primary={true} onClick={() => onSaveButtonClick(topic.id)}>
```

Comment:
Don't use lambdas in jsx props, since it [can cause unnecessary re-renders](https://stackoverflow.com/questions/36677733/why-shouldnt-jsx-props-use-arrow-functions-or-bind). Convert the component to a class instead and use [property initializers](https://www.fullstackreact.com/articles/use-property-initializers-for-cleaner-react-components/) to define the handlers. 

I made an issue to update .eslintrc to check for this type of problem; will try to get that done asap.